services/classic/management/http.go

```diff
@@ -100,22 +101,26 @@ func (client client) sendAzureRequest(method, url, contentType string, data []by
 }
 
 // createHTTPClient creates an HTTP Client configured with the key pair for
-// the subscription for this client.
+// the subscription for this client. If a client already exists, then it returns
+// the instance
 func (client client) createHTTPClient() (*http.Client, error) {
-	cert, err := tls.X509KeyPair(client.publishSettings.SubscriptionCert, client.publishSettings.SubscriptionKey)
-	if err != nil {
-		return nil, err
-	}
+	if client.httpClient == nil {
```

Comment:
Looks like this is only called from `sendAzureRequest()`.  Since you're now calling it in `makeClient()` I think you could remove the call from `sendAzureRequest()` and eliminate this nil check (also the receiver is passed by value so modifying it here doesn't have any effect).